### ISSUE TITLE
README directions on running the examples?

### ISSUE THREAD

**Issue by AndresCuervo** [MindBuffer/nannou#45]
It may be because it's been a while since I've used Rust, but I can't figure out how to run the examples in `/examples` - some build steps would be great.

I presume I'm doing something wrong because running just `cargo build` gives me a fatal error. As I work through this issues would love to help out building this project!

**Comment by mitchmindtree** [comment]
Hey thanks for the issue!

I think this is a great idea. I've just added PR #47 which adds a **Running Examples** section to the README. Is something like this what you were after?

> running just `cargo build` gives me a fatal error

Hmm sounds strange! Would you mind pasting the entire output of the error (including the command(s) used to produce it) here so I can take a look? If you could also include the output of `rustc --version` and `cargo --version` and your operating system (and version) that would be great.

**Comment by AndresCuervo** [comment]
Yep, the `release` command is the piece that's helpful, a command I could get just copy/paste to get the examples running :)

----

### Re: the fatal error

`cargo run --release --example simple_window` and `cargo build` both give me the following output:
```rust
   Compiling nannou v0.1.0 (file:///Users/andrescuervo/code/play/rust/nannou)
error[E0277]: the trait bound `for<'r> std::ops::FnMut(audio::cpal::cpal_impl::VoiceId, audio::cpal::UnknownTypeBuffer<'r>) + 'static: std::marker::Send` is not satisfied
   --> src/app.rs:261:18
    |
261 |                 .spawn(move || event_loop.run(move |v_id, out| loop_context.process(v_id, out)))
    |                  ^^^^^ `for<'r> std::ops::FnMut(audio::cpal::cpal_impl::VoiceId, audio::cpal::UnknownTypeBuffer<'r>) + 'static` cannot be sent between threads safely
    |
    = help: the trait `std::marker::Send` is not implemented for `for<'r> std::ops::FnMut(audio::cpal::cpal_impl::VoiceId, audio::cpal::UnknownTypeBuffer<'r>) + 'static`
    = note: required because of the requirements on the impl of `std::marker::Send` for `&'static mut for<'r> std::ops::FnMut(audio::cpal::cpal_impl::VoiceId, audio::cpal::UnknownTypeBuffer<'r>) + 'static`
    = note: required because of the requirements on the impl of `std::marker::Send` for `std::ptr::Unique<&'static mut for<'r> std::ops::FnMut(audio::cpal::cpal_impl::VoiceId, audio::cpal::UnknownTypeBuffer<'r>) + 'static>`
    = note: required because it appears within the type `alloc::raw_vec::RawVec<&'static mut for<'r> std::ops::FnMut(audio::cpal::cpal_impl::VoiceId, audio::cpal::UnknownTypeBuffer<'r>) + 'static>`
    = note: required because it appears within the type `std::vec::Vec<&'static mut for<'r> std::ops::FnMut(audio::cpal::cpal_impl::VoiceId, audio::cpal::UnknownTypeBuffer<'r>) + 'static>`
    = note: required because of the requirements on the impl of `std::marker::Send` for `std::sync::Mutex<std::vec::Vec<&'static mut for<'r> std::ops::FnMut(audio::cpal::cpal_impl::VoiceId, audio::cpal::UnknownTypeBuffer<'r>) + 'static>>`
    = note: required because it appears within the type `audio::cpal::cpal_impl::ActiveCallbacks`
    = note: required because of the requirements on the impl of `std::marker::Send` for `std::sync::Arc<audio::cpal::cpal_impl::ActiveCallbacks>`
    = note: required because it appears within the type `audio::cpal::cpal_impl::EventLoop`
    = note: required because it appears within the type `audio::cpal::EventLoop`
    = note: required because of the requirements on the impl of `std::marker::Send` for `std::sync::Arc<audio::cpal::EventLoop>`
    = note: required because it appears within the type `[closure@src/app.rs:261:24: 261:96 event_loop:std::sync::Arc<audio::cpal::EventLoop>, loop_context:audio::stream::output::LoopContext]`

error: aborting due to previous error

error: Could not compile `nannou`.
```

I'm on:

- cargo 0.23.0 (61fa02415 2017-11-22)
- rustc 1.22.1 (05e2e1c41 2017-11-22)
- MacOS 10.12.6 (Sierra)

**Comment by AndresCuervo** [comment]
While there are other generated errors, changing line 261 as noted above to:

```rust
 .spawn(|| { println!(""); })
```

You of course get a bunch of variable unused & `unnecessary mut` declaration warnings, but just as a proof of concept that this is the only line blocking my compilation step.

**Comment by mitchmindtree** [comment]
Ahh @AndresCuervo I think your problem relates to some CPAL macos backend issues that I believe were recently solved in tomaka/cpal#186 and tomaka/cpal#189. I've added a new PR that publishes a new version at tomaka/cpal#190 - once that is merged we can update nannou and hopefully get these fixes.

Going to close this issue now as the macos  cpal backend issues are a separate issue.

**Comment by AndresCuervo** [comment]
Ah, thanks for the info, will try again when that gets merged in then! :) Thanks @mitchmindtree
